lib/main.go

```diff
@@ -200,6 +200,16 @@ func (obj *Main) Exit(err error) {
 	obj.exit <- err // trigger an exit!
 }
 
+func (obj *Main) createTmpPrefix(prefix *string, hostname string) error {
+	var err error
+	if *prefix, err = ioutil.TempDir("", obj.Program+"-"+hostname+"-"); err != nil {
```

Comment:
this is a bit awkward, just return (string, error) instead.

Also, (and with admitted guilt) even though I've written bad code previously, if you're moving it around, you also get to fix it :) :) So, `obj.Program+"-"+hostname+"-"` should probably use `fmt.Sprintf` instead.